dvc/cli.py

```diff
@@ -102,11 +123,23 @@ def _find_parser(parser, cmd_cls):
 class DvcParser(argparse.ArgumentParser):
     """Custom parser class for dvc CLI."""
 
+    cmd_choices = []
+
     def error(self, message, cmd_cls=None):  # pylint: disable=arguments-differ
         logger.error(message)
         _find_parser(self, cmd_cls)
 
     def parse_args(self, args=None, namespace=None):
+        # NOTE: this is a custom check to see if any suggestions can
+        # be displayed to users in case of small typos
+        # E.g. `dvc commti` would display
+        # `The most similar command(s) are commit`
+        if args and args[0] not in self.cmd_choices:
```

Comment:
When DVC is run from the command line, args will be `None` here (meaning argparse should default to using `sys.argv`), so the suggested command code is never hit (except when we explicitly call `parse_args` with an argument list from the unit tests).

Reply:
We'll need to do the same kind of thing that [argparse](https://github.com/python/cpython/blob/38e32872eb3cf0dc9dd8cef9b05e47ba03638d34/Lib/argparse.py#L1774) does here:

```suggestion
        if args is None:
            args = sys.argv[1:]
        else:
            args = list(args)
        if args and args[0] not in self.cmd_choices:
```